components/net_traits/pub_domains.rs

```diff
@@ -146,11 +146,13 @@ pub fn is_reg_domain(domain: &str) -> bool {
 /// Returns None if the URL has no host name.
 /// Returns the registered suffix for the host name if it is a domain.
 /// Leaves the host name alone if it is an IP address.
-pub fn reg_host<'a>(url: &'a ServoUrl) -> Option<Host> {
+pub fn reg_host(url: &ServoUrl) -> Option<Host> {
     match url.origin() {
-        ImmutableOrigin::Tuple(_, Host::Domain(domain), _) => Some(Host::Domain(String::from(reg_suffix(&*domain)))),
-        ImmutableOrigin::Tuple(_, Host::Ipv4(address), _) => Some(Host::Ipv4(address)),
-        ImmutableOrigin::Tuple(_, Host::Ipv6(address), _) => Some(Host::Ipv6(address)),
+        ImmutableOrigin::Tuple(_, host, _) => if let Host::Domain(domain) = host {
+            Some(Host::Domain(String::from(reg_suffix(&*domain))))
+        } else {
+            Some(host)
+        },
```

Comment:
This is worse than before or what I suggested and the formatting is wrong. Don't nest an `if let` expression inside a match arm without enclosing it in braces.